src/metadata_validator.py

```diff
@@ -345,17 +346,56 @@ def validate_relationship(self, data_record):
                 result[ERRORS].append(create_error("Invalid relationship", f'Property “{parent_id_property}" of related node “{parent_type}” is empty.'))
                 continue
 
-            if (parent_type, parent_id_property, parent_id_value) not in parent_node_cache:
+            rel_type = node_relationships[parent_type].get(TYPE)
+            # check if there is only one parent for both one_to_one and many_to_one relationships.
+            if rel_type != "many_to_many": 
+                if not checked_multi_parents:
+                    parents = [item for item in data_record_parent_nodes if item[PARENT_TYPE] == parent_type and item[PARENT_ID_NAME] == parent_id_property ]
+                    if len(parents) > 1:
+                        error_type = "One-to-one relationship conflict" if rel_type == "one_to_one" else "Many-to-one relationship conflict"
+                        parent_node_ids = [item[PARENT_ID_VAL] for item in parents]
+                        result[ERRORS].append(create_error(error_type, 
+                                    f'"{msg_prefix}": associated with multiple “{parent_type}” nodes: {json.dumps(parent_node_ids)}.'))
+                    checked_multi_parents = True
```

Comment:
If current node has two relationships, one of them one-to-one, the other many-to-one, will **checked_multi_parents** correctly track both?

Reply:
updated.